### PR TITLE
Fixing exposing parsing

### PR DESCRIPTION
**Pull request by eeue56**
Outfoxes https://github.com/rtfeldman/node-test-runner/issues/136

Also adds a whole bunch of tests for the parser. If I get the chance, there's a refactoring potential to put the test searching into this file too - making for less filesystem locks

### CODE REVIEW

**Comment by rtfeldman** [comment]
😻 😻 😻 😻 😻

**Comment by rtfeldman** [comment]
Thanks @eeue56 - looks great!

Also looks familiar...this is based on your webpack loader optimization work, yeah?

File: lib/finder.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      return;
+    }
+
+    // if we are still reading the module line
+    if (isReadingModuleName) {
+      var exposingIndex = line.indexOf('exposing');
+
+      // if we haven't found exposing yet
+      if (exposingIndex === -1) {
+        if (line.length > 0) parsingDone = true;
```
Comment (zwilias) [inline lib/finder.js]: This seems a little weird, and I think it's why #138 is occurring. @eeue56 could you clarify in which case this is required?

Comment (eeue56) [inline lib/finder.js]: on the money

**Pull request merged**